apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/_components/Sidebar/index.tsx

```diff
@@ -1,29 +1,56 @@
 import {
-  Commit,
-  DocumentVersion,
+  CommitsRepository,
+  CommitStatus,
   DocumentVersionsRepository,
 } from '@latitude-data/core'
+import { Commit, DocumentVersion, Project } from '@latitude-data/core/browser'
 import { DocumentSidebar } from '@latitude-data/web-ui'
+import { fetchCommitsByProjectAction } from '$/actions/commits/fetchCommitsByProjectAction'
 import { getCurrentUser } from '$/services/auth/getCurrentUser'
 
 import ClientFilesTree from './ClientFilesTree'
+import CommitSelector from './CommitSelector'
 
 export default async function Sidebar({
+  project,
   commit,
   currentDocument,
 }: {
+  project: Project
   commit: Commit
   currentDocument?: DocumentVersion
 }) {
   const { workspace } = await getCurrentUser()
   const docsScope = new DocumentVersionsRepository(workspace.id)
   const documents = await docsScope.getDocumentsAtCommit(commit)
+  const [draftCommits, fetchCommitsError] = await fetchCommitsByProjectAction({
+    projectId: project.id,
+    status: CommitStatus.Draft,
+  })
+  const commitsScope = new CommitsRepository(workspace.id)
+  const headCommit = await commitsScope
+    .getHeadCommit(project)
+    .then((r) => r.unwrap())
+
+  if (fetchCommitsError) {
+    throw fetchCommitsError
+  }
+
   return (
-    <DocumentSidebar>
-      <ClientFilesTree
-        currentDocument={currentDocument}
-        documents={documents.unwrap()}
-      />
-    </DocumentSidebar>
+    <DocumentSidebar
+      header={
+        <CommitSelector
+          headCommitId={headCommit.id}
+          currentCommit={commit}
+          draftCommits={draftCommits}
+        />
+      }
+      tree={
+        <ClientFilesTree
+          currentDocument={currentDocument}
```

Comment:
hmm minor nitpick you usually want this to be an id and fetch the current document from the documents array within the ClientFilesTree component so to avoid duplicated data sync issues (stuff like currentDocument not being present in documents array, or currentDocument having different data than document).

Reply:
Indeed this object is only pass down to do Id comparison